tests/test_ancillaries.py

```diff
@@ -72,6 +109,64 @@ def test_af_deform():
     assert np.allclose(ds["deform"], 1 - ds["circ"])
 
 
+@pytest.mark.parametrize("path", example_data_sets)
+def test_af_method_called_once_with_shared_pipeline(
+        fake_af_cleanup_fixture, path):
+    """Verifies that when an `AncillaryFeature.method` is shared between
+    ancillary features, the `method` itself is only called once. All ancillary
+    features that share the `method` will be populated in this single call.
+    """
+    af1, af2 = fake_af_cleanup_fixture
+    ds = dclab.new_dataset(retrieve_data(path))
+    assert af1.feature_name not in ds.features_innate
+    assert af2.feature_name not in ds.features_innate
+    assert af1.feature_name in ds
+    assert af2.feature_name in ds
+
+    # handle tdms "time" `AncillaryFeature`
+    tdms_time_feature = False
+    feats = [af1.feature_name, af2.feature_name]
+    if "tdms" in ds.identifier:
```

Comment:
You can use `ds.format == "tdms"` here (https://github.com/ZELLMECHANIK-DRESDEN/dclab/blob/master/dclab/rtdc_dataset/core.py#L37).